torch/distributed/_tensor/dispatch.py

```diff
@@ -230,6 +238,26 @@ def default_tensor(spec: DTensorSpec) -> torch.Tensor:
             redistribute_with_schema=needs_redistribute,
         )
 
+        aten = torch.ops.aten
+        random_ops = [
+            aten.native_dropout.default,
+            aten.normal_.default,
+            aten.uniform_.default,
+        ]
+        # before running local op computation, check if op is random op
+        # for random ops, set RNG offset
+        if op_call in random_ops:
+            assert isinstance(mesh, DeviceMesh)
```

Comment:
hmmm I don't think we need to add those asserts, it's pretty certain that the `mesh` is a DeviceMesh as we did the checks previously in L143.

Reply:
Not really. The previous check is `if mesh is not None and mesh.get_coordinate() is None` so entering the `else` branch does not guarantee that `mesh` is not a `None` object. Removing `assert isinstance(mesh, DeviceMesh)` causes lint error.

Reply:
Ok that make sense